zerver/tests/test_subs.py

```diff
@@ -1638,10 +1648,24 @@ def test_non_ascii_stream_subscription(self) -> None:
         """
         Subscribing to a stream name with non-ASCII characters succeeds.
         """
-        self.helper_check_subs_before_and_after_add(self.streams + [u"hümbüǵ"], {},
+        self.helper_check_subs_before_and_after_add(self.streams + [u"hümbüǵ"],
+                                                    dict(principals=ujson.dumps([self.example_email("hamlet")])),
                                                     [u"hümbüǵ"], self.streams, self.test_email,
                                                     self.streams + [u"hümbüǵ"], self.test_realm)
 
+    def test_subscriptions_with_zero_principals(self) -> None:
+        """
+        Stream must have at least one subscriber.
+        """
+        private_stream_name = "private_stream_with_zero_subs"
+        result = self.common_subscribe_to_streams(self.test_email, [private_stream_name],
+                                                  dict(principals=ujson.dumps([])))
+        self.assert_json_error(result, "Stream must have at least one subscriber.")
+        public_stream_name = "public_stream_with_zero_subs"
+        result = self.common_subscribe_to_streams(self.test_email, [public_stream_name],
+                                                  dict(principals=ujson.dumps([])))
+        self.assert_json_error(result, "Stream must have at least one subscriber.")
```

Comment:
Unless I am mistaken, this test would be redundant after you read my comment below. :)